src/hacbs/components/ImportForm/utils/submit-utils.ts

```diff
@@ -0,0 +1,30 @@
+import { FormikHelpers } from 'formik';
+import { createComponents } from '../../../../components/ImportForm/utils/submit-utils';
+import { ImportFormValues } from '../../../../components/ImportForm/utils/types';
+import { createApplication } from '../../../../utils/create-utils';
+import { FormValues } from '../types';
+
+export const onComponentsSubmit = async (
+  { components, applicationData, namespace, secret }: FormValues,
+  formikBag: FormikHelpers<ImportFormValues>,
+) => {
+  try {
+    await createComponents(components, applicationData.metadata.name, namespace, secret, true);
+  } catch (error) {
+    const message = error.code === 409 ? 'Component name already exists.' : error.message;
+    formikBag.setStatus({ submitError: message });
+    throw error;
+  }
+};
+
+export const onApplicationSubmit = async ({ application, namespace }: FormValues, formikBag) => {
+  try {
+    await createApplication(application, namespace, true);
+    const applicationData = await createApplication(application, namespace, false);
+    formikBag.setFieldValue('applicationData', applicationData);
+  } catch (error) {
+    const message = error.code === 409 ? 'Application name already exists.' : error.message;
+    formikBag.setFieldError('application', message);
```

Comment:
App studio sets the form status instead of a field error.
Also the status is set for component creation instead of the field error.

We should be consistent. So which one?

Reply:
Thanks for raising this @christianvogt 

The inline alert for the Application name is correct. We should follow that convention for the whole flow since the user needs guidance in what they need to fix. For the Component name error, I suggest we also use an inline alert and put the field into edit mode when the error is returned. So the user does not even have to click the pencil.